backend/apps/owasp/management/commands/create_project_metadata_file.py

```diff
@@ -0,0 +1,75 @@
+"""A command to generate project metadata YAML files."""
+
+from apps.owasp.management.commands.base_generate_metadata import BaseGenerateMetadataCommand
+from apps.owasp.models.project import Project, ProjectLevel, ProjectType
+
+MIN_PROJECT_TAGS = 3
+
+
+class Command(BaseGenerateMetadataCommand):
+    help = "Generates and validates project metadata for a given project key."
+    model = Project
+    schema_name = "project"
+
+    def map_data_to_schema(self, project: Project) -> dict:
+        """Map the Project model data to a dictionary that matches the schema."""
+        level_mapping = {
+            ProjectLevel.INCUBATOR: 2,
+            ProjectLevel.LAB: 3,
+            ProjectLevel.PRODUCTION: 3.5,
+            ProjectLevel.FLAGSHIP: 4,
+        }
+        type_mapping = {
+            ProjectType.CODE: "code",
+            ProjectType.DOCUMENTATION: "documentation",
+            ProjectType.TOOL: "tool",
```

Comment:
Please refactor existing code to have a single source of truth for these mappings.